### ISSUE TITLE
http-axum example not working with version 1.0 events

### ISSUE THREAD

**Issue by hxk1633** [awslabs/aws-lambda-rust-runtime#702]
I'm trying to invoke lambdas using the REST API Gateway and the calls keep calling empty 404 errors. It works if I switch from Api to HttpApi events in the sam template.yaml file. It appears that passing in the axum router as shown in the http-axum example in this repo doesn't work for version 1.0 events that REST API Gateway uses.

Cargo dependencies:
lambda_http = {version = "0.8.1", features = ["apigw_rest"], default-features = false}
lambda_runtime = "0.8.1"
axum = "0.6.4"
tokio = { version = "1", features = ["macros"] }

**Comment by bnusunny** [comment]
For events from HTTP API trigger, lambda-http only support version 2.0 format. Could you please update the event format on HTTP API?

**Comment by hxk1633** [comment]
> For events from HTTP API trigger, lambda-http only support version 2.0 format. Could you please update the event format on HTTP API?

I'm not sure what you mean by updating the event format. REST API Gateway only supports v1 events while the HTTP API supports v2. The docs say that lambda-http supports both types of API Gateways.

**Comment by bnusunny** [comment]
REST API -> event format version 1
HTTP API -> event format version 2

**Comment by hxk1633** [comment]
As I mentioned in the original post, HttpApi events (version 2.0, HTTP API) are working as expected for me. I looked at the --debug logs and they are sending version 2.0 event payloads to the lambda. When I switch to Api events in sam template, it's sending version 1.0 event payloads which causes the REST API Gateway to return 404 error for all routes I hit.

**Comment by bnusunny** [comment]
Did you check the REST API Endpoint URL? It has a stage in the path.

**Comment by hxk1633** [comment]
Yeah I always include the stage in the path when deploying to aws. This issue also occurs when I run the api gateway locally using `sam local start-api`

**Comment by bnusunny** [comment]
Hmm, the path axum receives contains the stage. I not sure when this changed. 

@calavera @greenwoodcm Do you recall when REST API events has stage added to the path? It seems to break http-axum example.

**Comment by hxk1633** [comment]
After digging through the source code (request.rs) and adding print statements, I found out the default stage name being added to the path is 'Prod' for the rest api gateway. So I added this to my axum routes and it started to work as expected. Changing the axum routes is not required for the http api gateway since it passes in a stage with a value of either '$default' or None.

**Comment by wduminy** [comment]
@hxk1633 I have the same issue.  This call works `curl --request GET --url http://localhost:9000/foo` 
But `cargo lambda invoke lambda --data-file ../scripts/api-gw.json`

**Comment by greenwoodcm** [comment]
I wasn't aware this changed

**Comment by ramunecroft** [comment]
same problem

**Comment by JensGM** [comment]
Any update on this? We need to use the REST API, so version 1
